examples/python/controlnet/main.py

```diff
@@ -0,0 +1,141 @@
+    rr.log("output", rr.Image(image))
+    rr.log("latent", rr.Tensor(latents[0], dim_names=["channel", "height", "width"]))
+
+
+def run_canny_controlnet(image_path: str, prompt: str, negative_prompt: str):
+    if image_path.startswith("http://") or image_path.startswith("https://"):
+        pil_image = PIL.Image.open(requests.get(image_path, stream=True).raw)
+    elif os.path.isfile(image_path):
+        pil_image = PIL.Image.open(image_path)
+    else:
+        raise ValueError(f"Invalid image_path: {image_path}")
+
+    image = np.array(pil_image)
+
+    if image.shape[2] == 4:  # RGBA image
+        rgb_image = image[..., :3]  # RGBA to RGB
+        rgb_image[image[..., 3] < 200] = 0.0  # reduces artifacts for transparent parts
+    else:
+        rgb_image = image
+
+    low_threshold = 100.0
+    high_threshold = 200.0
+    canny_image = cv2.Canny(rgb_image, low_threshold, high_threshold)
+    canny_image = canny_image[:, :, None]
+    # cv2.dilate(kjgk
+    canny_image = np.concatenate([canny_image, canny_image, canny_image], axis=2)
+    canny_image = PIL.Image.fromarray(canny_image)
+
+    rr.log("input/raw", rr.Image(image), timeless=True)
+    rr.log("input/canny", rr.Image(canny_image), timeless=True)
+
+    controlnet = ControlNetModel.from_pretrained(
+        "diffusers/controlnet-canny-sdxl-1.0",
+        torch_dtype=torch.float16,
+        use_safetensors=True,
+    )
+    vae = AutoencoderKL.from_pretrained(
+        "madebyollin/sdxl-vae-fp16-fix", torch_dtype=torch.float16, use_safetensors=True
+    )
+    pipeline = StableDiffusionXLControlNetPipeline.from_pretrained(
+        "stabilityai/stable-diffusion-xl-base-1.0",
+        controlnet=controlnet,
+        vae=vae,
+        torch_dtype=torch.float16,
+        use_safetensors=True,
+    )
+    pipeline.enable_model_cpu_offload()
+
+    rr.log(
+        "positive_prompt",
+        rr.TextDocument(f"### Positive Prompt\n {prompt}", media_type="text/markdown"),
+        timeless=True,
+    )
+    rr.log(
+        "negative_prompt",
+        rr.TextDocument(
+            f"### Negative Prompt\n {negative_prompt}", media_type="text/markdown"
+        ),
+        timeless=True
+    )
+
+    images = pipeline(
+        prompt,
+        negative_prompt=negative_prompt,
+        image=canny_image,  # add batch dimension
+        controlnet_conditioning_scale=0.5,
+        callback=lambda i, t, latents: controlnet_callback(i, t, latents, pipeline),
+    ).images[0]
+
+    rr.log("output", rr.Image(images))
+
+
+def main() -> None:
+    parser = argparse.ArgumentParser(
+        description="Use Canny-conditioned ControlNet to generate image."
+    )
+    parser.add_argument(
+        "--img_path",
+        type=str,
+        help="Path to image used as input for Canny edge detector.",
+        default="https://huggingface.co/datasets/hf-internal-testing/diffusers-images/resolve/main/sd_controlnet/hf-logo.png",
```

Comment:
```suggestion
        default="https://storage.googleapis.com/rerun-example-datasets/controlnet/rerun-icon-512.png",
```

Reply:
Actually, this should probably be broken out to a constant

Reply:
Done

Reply:
From quite limited testing results seem to be much better with the high res logo, so I'm using that as the default.